### PR TITLE
Added AuthenticatedDevice

### LINKED ISSUE #32: Set up authentication on endpoints

### PR DESCRIPTION
**Pull request by UnluckyBird**
Added authentication, which can be added to any endpoint
Closes #32

### CODE REVIEW

**Review by Laursen79 — COMMENTED** [review]
Mads skal nok lige se på SQL

**Review by MJHC — DISMISSED** [review]
ok

File: server/database/init.sql [hunk 1]
```diff
@@ -1,8 +1,8 @@
 -- TABLES
 CREATE TABLE accounts (
     id                INTEGER PRIMARY KEY NOT NULL GENERATED ALWAYS AS IDENTITY,
-    aci               VARCHAR(16) NOT NULL UNIQUE,
-    pni               VARCHAR(20) NOT NULL UNIQUE,
+    aci               TEXT NOT NULL UNIQUE,
+    pni               TEXT NOT NULL UNIQUE,
```
Comment (Darkros1245) [inline server/database/init.sql:5]: These should be fixed length

File: server/database/init.sql [hunk 2]
```diff
@@ -12,7 +12,7 @@ CREATE TABLE devices (
     owner       INTEGER NOT NULL REFERENCES accounts(id) ON DELETE CASCADE,
     device_id   TEXT NOT NULL,
     name        BYTEA NOT NULL,
-    auth_token  VARCHAR(255) NOT NULL UNIQUE,
+    auth_token  BYTEA NOT NULL,
```
Comment (Darkros1245) [inline server/database/init.sql:15]: This should also, if not already, be changed in the quries

File: server/src/postgres.rs [hunk 3]
```diff
@@ -66,9 +66,8 @@ impl SignalDatabase for PostgresDatabase {
             FROM
                 accounts
             WHERE
-                $1 = $2
+                aci = $1 OR pni = $1
```
Comment (Darkros1245) [inline server/src/postgres.rs:69]: I have already made these changes for this file

**Review by Darkros1245 — CHANGES_REQUESTED** [review]

File: server/database/init.sql (hunk 2 above)
Comment (UnluckyBird) [reply server/database/init.sql:16]: Already was changed

File: server/database/init.sql (hunk 1 above)
Comment (UnluckyBird) [reply server/database/init.sql]: Fixed

File: server/src/postgres.rs (hunk 3 above)
Comment (UnluckyBird) [reply server/src/postgres.rs]: Fixed in merge conflicts

**Review by Darkros1245 — APPROVED** [review]

**Review by MJHC — APPROVED** [review]
ok

**Pull request merged**